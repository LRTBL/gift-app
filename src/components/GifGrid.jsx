import React from 'react';
import PropTypes from 'prop-types';
import useFetchGifs from '../hooks/useFetchGifs';
import GifGridIitem from './GifGridIitem';

const GifGrid = ({ category }) => {
    const { data: images, loading } = useFetchGifs(category);

    return (
        <>
            {loading && <p className='animate__animated animate__flash'>Cargando ...</p>}
            <h3 className='animate__animated animate__fadeIn'>{category}</h3>
            <div className='card-grid'>
                {images.map((img) => (
                    <GifGridIitem key={img.id} img={img} />
                ))}
            </div>
        </>
    );
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
};

export default GifGrid;
