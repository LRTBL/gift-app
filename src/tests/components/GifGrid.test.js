/* eslint-disable no-undef */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import GifGrid from '../../pages/gift-expert/components/GifGrid';
import useFetchGifs from '../../pages/gift-expert/hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

const { shallow } = require('enzyme');

describe('Pruebas de <GifGrid/>', () => {
    const category = 'One Punch';

    test('Debe de mostrar el componente correctamente', () => {
        const gifs = [
            {
                id: 'ABC',
                url: 'hhtp',
                title: 'Cualquier cosa',
            },
        ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar items cuando se cargar imÃ¡genes con useFetchGifs', () => {
        const gifs = [
            {
                id: 'ABC',
                url: 'hhtp',
                title: 'Cualquier cosa',
            },
            {
                id: '123',
                url: 'hhtp',
                title: 'Cualquier cosa',
            },
        ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        const wrapper = shallow(<GifGrid category={category} />);
        // expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
});
