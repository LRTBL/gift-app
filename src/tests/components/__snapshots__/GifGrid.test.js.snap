// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifGrid/> Debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="hhtp"
    />
  </div>
</Fragment>
`;

exports[`Pruebas de <GifGrid/> Debe de mostrar items cuando se cargar im√°genes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="hhtp"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Cualquier cosa"
      url="hhtp"
    />
  </div>
</Fragment>
`;
